website/docs/r/quicksight_ip_restriction.html.markdown

```diff
@@ -0,0 +1,62 @@
+---
+subcategory: "QuickSight"
+layout: "aws"
+page_title: "AWS: aws_quicksight_ip_restriction"
+description: |-
+  Manages the content and status of IP rules.
+---
+
+# Resource: aws_quicksight_ip_restriction
+
+Manages the content and status of IP rules.
+
+~> Deletion of this resource clears all IP restrictions from a QuickSight account.
+
+## Example Usage
+
+```terraform
+resource "aws_quicksight_ip_restriction" "example" {
+  enabled = true
+
+  ip_restriction_rule_map = {
+    "108.56.166.202/32" = "Allow self"
+  }
+
+  vpc_id_restriction_rule_map = {
+    (aws_vpc.example.id) = "Main VPC"
```

Comment:
Are the parenthesis necessary here?

Reply:
Yes, otherwise `terraform plan` reports

```
╷
│ Error: Ambiguous attribute key
│ 
│   on main.tf line 15, in resource "aws_quicksight_ip_restriction" "test":
│   15:     aws_vpc.test.id = "Main VPC"
│ 
│ If this expression is intended to be a reference, wrap it in parentheses. If it's instead intended as a literal name containing periods, wrap it
│ in quotes to create a string literal.
╵
```

Reply:
TIL - thanks!